qemu/tests/chardev_free_port.py

```diff
@@ -0,0 +1,46 @@
+from virttest import error_context, env_process
+from qemu.tests.virtio_console import add_chardev
+from virttest import utils_misc
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    qemu should try to find a free port value by to=<port> with unix socket and tcp options:
+    1) boot guest with socket with -qmp tcp:0:4444.
+    2) telnet localhost.
+    3) check output of {"execute": "qmp_capabilities"}, {"execute":"query-chardev"}.
+    4) repeat step 1 to 3 with -qmp tcp:0:4445 and telnet to port 4445.
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    chardev_infos = []
+    port_nums = []
+    vms = ['vm1', 'vm2']
+    for ind, vm in enumerate(vms):
+        vm1 = env.get_vm(vm)
+        char_devices = add_chardev(vm1, params)
+        chardev_params = char_devices[0].params
+        if ind == 1:
+            host = chardev_params['host']
+            chardev_to = utils_misc.find_free_ports(int(port_nums[0]) + 1, 6000, 1, host)
+            params['chardev_to_channel1'] = chardev_to
+
+        for device in char_devices:
+            extra_params = ' ' + device.cmdline()
+            params['extra_params_%s' % vm] = params.get('extra_params', '') + extra_params
+        params['start_vm_%s' % vm] = "yes"
+        env_process.preprocess(test, params, env)
+        vm1 = env.get_vm(vm)
+        chardev_infos.append(vm1.monitor.info("chardev"))
+        port_nums.append(chardev_params['port'])
+
+    for ind, chardevs in enumerate(chardev_infos):
+        in_chardev = False
+        for chardev in chardevs:
+            if port_nums[ind] in chardev['filename']:
+                in_chardev = True
+                break
+        assert in_chardev is True
```

Comment:
use `test.fail` and add special error message.

Reply:
At least, add message when assert failed:
`assert in_cahrdev is True, error_message_to_explain_this_fail`